### PR TITLE
TOK-455

### PR DESCRIPTION
**Pull request by mattraffel**
Cases to test
1) input hasn't been set totally
2) inexistent payment-source
3) insufficient funds
4) extra funds

### CODE REVIEW

File: libsovtoken/tests/payment_chaos_tests.rs [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+        mint_tokens: None,
+        fees: None,
+    });
+    let payment_addresses = &setup.addresses;
+    let pool_handle = setup.pool_handle;
+    let dids = setup.trustees.dids();
+    let tokens : i32 = 50;
+
+    // ---- create some tokens
+    do_minting(pool_handle, &wallet, &dids, &payment_addresses[0], tokens);
```
Comment (dastardlychimp) [inline libsovtoken/tests/payment_chaos_tests.rs]: You can do minting as part of the Setup/SetupConfig, so you don't need to have this do_minting function.

File: libsovtoken/tests/payment_chaos_tests.rs [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+    let pay_output_json = json!([
+        {
+            "recipient": payment_addresses[0],
+            "amount": 50
+        }
+    ]).to_string();
+
+    match indy::payments::Payment::build_payment_req(wallet.handle, Some(dids[0]), &pay_input_json, &pay_output_json, None) {
+        Ok(_) => { assert!(false, "Expected CommonInvalidStructure error"); }
+        Err(ec) => { assert_eq!(ec, ErrorCode::CommonInvalidStructure); }
```
Comment (dastardlychimp) [inline libsovtoken/tests/payment_chaos_tests.rs]: You could use an assert_eq! assertion which is a little more condensed.

```
let ec = (...).unwrap_err();
assert_eq!(ErrorCode::CommonInvalidStructure, ec);
```
or
```
assert_eq!(Err(ErrorCode::CommonInvalidStructure), result);
```

Comment (mattraffel) [reply libsovtoken/tests/payment_chaos_tests.rs]: much better.  thnx

File: libsovtoken/tests/payment_chaos_tests.rs [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+use indy::ErrorCode;
+use utils::setup::{Setup, SetupConfig};
+use utils::wallet::Wallet;
+
+// ------------------------------------------------------------------------------------------------
+#[test]
+pub fn pay_without_outputs_fails() {
+
+    // ---- setup
+    sovtoken::api::sovtoken_init();
```
Comment (dastardlychimp) [inline libsovtoken/tests/payment_chaos_tests.rs]: This is called inside `Setup::new` and doesn't need to be called if you use the Setup function.

Comment (cam-parra) [reply libsovtoken/tests/payment_chaos_tests.rs]: Has this been resolved?

Comment (mattraffel) [reply libsovtoken/tests/payment_chaos_tests.rs]: it has not.  good catch.  I will address that now.

Comment (mattraffel) [reply libsovtoken/tests/payment_chaos_tests.rs]: latest change addresses this

**Pull request merged**